### PR TITLE
Add Casual Game Engine

### PR DESCRIPTION
**Pull request by MichaelHinrichs**
### SteamDB app page links to a few games using this
https://steamdb.info/app/1725730/
https://steamdb.info/app/1001860/
https://steamdb.info/app/1807240/
https://steamdb.info/app/1653360/
https://steamdb.info/app/1858860/

### Brief explanation of the change
Find Casual Game Engine by it's AngelScript files.

### CODE REVIEW

File: rules.ini [hunk 1]
```diff
@@ -43,6 +43,7 @@ BlenderGameEngine[] = (?:^|/)BlendThumb(?:64)?\.dll$
...
 Build = (?:^|/)buildlic\.txt$
 C4_Engine = (?:^|/)Data/C4(?:\.pak)?$
+CasualGameEngine = (?:^|/)(?:aircraft|asteroid|bazooka|bee|bomb|chainsaw|colorgun|defense_station|defense_station_cls|demo_sdk|fireworks|hhg|hidenseek|lasergun|lasermech|mp5|nuke|pistol|planet|plasma_ball|satellite|smoke|spawn_portal|stamp|vortex|(?:gunn|hamm|play|rlaunch|snip|soldi|(?:flamethr|meteorsh)ow)er|item_(?:ammo|ammo_(?:circlepulse|fthrower|grenade|gun|handgun|laser|laserball|lightlaser|mine|missile|plasmagun|rifle|shotgun)|coin|gem|health)|weapon_(?:arrow|bolt|circlepulse|flame|grenade|gun|mine|missile|(?:light)?laser|(?:laser|plasma)ball)|world_(?:events|obstacle|obstacle_(?:goo|lava)|portal|quests|shop|theme|wave(?:monitor|point(?:cls)?))|(?:alien(?:boltship|boss|infantry|lasership|lightlasership|missileship|plasmaship|rotatorship|tankship|vehicle)|ballista|cat|dog|fox|frogator|goat|headcrab|helicopter|mech|tank|teslatower|water(?:boltship|fighter|launcher)|wolfdragon)?(?:cls)?|bigexplosion|explosion(?:_destroy)?|poop(?:expl)?|(?:blood)?decal|(?:team)?versus|(?:bolt)?wand|(?:event|info|mapselect|quest|reward|shop|waveinfo)menu|food_(?:apple|orange|pear|strawberry)|tc_(?:dualgun|heavy|light))\.as$
```
Comment (Lifeismana) [inline rules.ini:46]: reduce the size of this regex
for what it's worth, here's the list of depots that have a file that ends with .as
* [17501 - Zombie Panic! Source Content](https://steamdb.info/depot/17501/)
* [18801 - ZeroGearDepot](https://steamdb.info/depot/18801/)
* [25002 - Overgrowth LinuxWinDataDepot](https://steamdb.info/depot/25002/)
* [25003 - Overgrowth MacAppDepot](https://steamdb.info/depot/25003/)
* [33671 - Disciples 3 Main](https://steamdb.info/depot/33671/)
* [43191 - The Haunted: Hells Reach Content](https://steamdb.info/depot/43191/)
* [43211 - the-haunted-hells-reach-dedicated content](https://steamdb.info/depot/43211/)
* [57611 - disciples III demo content](https://steamdb.info/depot/57611/)
* [70901 - star ruler content](https://steamdb.info/depot/70901/)
* [70911 - star ruler demo content](https://steamdb.info/depot/70911/)
* [100981 - 3Dcoat Content](https://steamdb.info/depot/100981/)
* [104311 - RedOrchestra2SDK1](https://steamdb.info/depot/104311/)
* [200671 - Disciples III Resurrection Main](https://steamdb.info/depot/200671/)
* [205793 - Source 2 Dota 2 Test Win64 Content](https://steamdb.info/depot/205793/)
* [209342 - Ride to Hell: Retribution Binaries](https://steamdb.info/depot/209342/)
* [215281 - The Secret World Content](https://steamdb.info/depot/215281/)
* [219641 - Chivalry: Medieval Warfare Content](https://steamdb.info/depot/219641/)
* [219831 - King Arthur's Gold Win32](https://steamdb.info/depot/219831/)
* [219832 - King Arthur's Gold OSX](https://steamdb.info/depot/219832/)
* [219833 - King Arthur's Gold Linux](https://steamdb.info/depot/219833/)
* [220981 - CRYENGINE Content](https://steamdb.info/depot/220981/)
* [225841 - Base content](https://steamdb.info/depot/225841/)
* [226721 - Lost Planet 3 Content](https://steamdb.info/depot/226721/)
* [227282 - No Time Mac](https://steamdb.info/depot/227282/)
* [232211 - Chivalry: Medieval Warfare Beta Content](https://steamdb.info/depot/232211/)
* [232811 - windows depot](https://steamdb.info/depot/232811/)
* [232812 - mac depot](https://steamdb.info/depot/232812/)
* [238321 - Outlast](https://steamdb.info/depot/238321/)
* [238432 - Contagion Client Shared Base Content](https://steamdb.info/depot/238432/)
* [241280 - Chivalry: Deadliest Warrior (241280) Depot](https://steamdb.info/depot/241280/)
* [243801 - Gas Guzzlers Extreme Content](https://steamdb.info/depot/243801/)
* [246721 - Base Content](https://steamdb.info/depot/246721/)
* [248251 - Windows Content](https://steamdb.info/depot/248251/)
* [249951 - UntitledApp Content](https://steamdb.info/depot/249951/)
* [249952 - Forge Quest OSX Depot](https://steamdb.info/depot/249952/)
* [249953 - Forge Quest Linux Depot](https://steamdb.info/depot/249953/)
* [249954 - Forge Quest Linux 32 Bit Depot](https://steamdb.info/depot/249954/)
* [251772 - Shadow Man Windows English Content](https://steamdb.info/depot/251772/)
* [251773 - Shadow Man Windows French Content](https://steamdb.info/depot/251773/)
* [251774 - Shadow Man Windows German Content](https://steamdb.info/depot/251774/)
* [251775 - Shadow Man Windows Italian Content](https://steamdb.info/depot/251775/)
* [251776 - Shadow Man Windows Spanish Content](https://steamdb.info/depot/251776/)
* [251777 - Shadow Man Mac English Content](https://steamdb.info/depot/251777/)
* [251778 - Shadow Man Mac French Content](https://steamdb.info/depot/251778/)
* [251779 - Shadow Man Mac German Content](https://steamdb.info/depot/251779/)
* [251780 - Shadow Man Mac Italian Content](https://steamdb.info/depot/251780/)
* [251781 - Shadow Man Mac Spanish Content](https://steamdb.info/depot/251781/)
* [252231 - Yaiba: Ninja Gaiden Z Content](https://steamdb.info/depot/252231/)
* [262101 - Trench Run Win32](https://steamdb.info/depot/262101/)
* [262102 - Trench Run OSX](https://steamdb.info/depot/262102/)
* [262103 - Trench Run Linux](https://steamdb.info/depot/262103/)
* [265101 - Gas Guzzlers Extreme Demo Content](https://steamdb.info/depot/265101/)
* [265551 - Walking Content](https://steamdb.info/depot/265551/)
* [267653 - Flash Exporter for Clickteam Fusion 2.5 (267653) Depot](https://steamdb.info/depot/267653/)
* [268671 - BaseContent](https://steamdb.info/depot/268671/)
* [270151 - Windows Content](https://steamdb.info/depot/270151/)
* [270152 - Linux Content](https://steamdb.info/depot/270152/)
* [270153 - RUNNING WITH RIFLES Mac Depot](https://steamdb.info/depot/270153/)
* [275311 - Victorian Admirals Content](https://steamdb.info/depot/275311/)
* [282591 - Star Ruler 2 Content](https://steamdb.info/depot/282591/)
* [286881 - Disney Planes Content](https://steamdb.info/depot/286881/)
* [288331 - RUNNING WITH RIFLES Demo Content Windows](https://steamdb.info/depot/288331/)
* [288332 - RUNNING WITH RIFLES Demo Depot Linux](https://steamdb.info/depot/288332/)
* [288333 - RUNNING WITH RIFLES Demo Depot Mac](https://steamdb.info/depot/288333/)
* [291571 - Rising Storm Beta SDK Content](https://steamdb.info/depot/291571/)
* [293031 - Game Base Content](https://steamdb.info/depot/293031/)
* [296771 - Real Boxing™ Depot](https://steamdb.info/depot/296771/)
* [301831 - Brave Content](https://steamdb.info/depot/301831/)
* [311291 - SpellForce 3 Content](https://steamdb.info/depot/311291/)
* [312862 - Defense Grid 2: A Matter of Endurance Content - Linux](https://steamdb.info/depot/312862/)
* [314715 - Mighty No. 9 Windows](https://steamdb.info/depot/314715/)
* [323851 - Move or Die Content](https://steamdb.info/depot/323851/)
* [332261 - Cars Mater-National Common Content](https://steamdb.info/depot/332261/)
* [332761 - Friendship Club (Windows)](https://steamdb.info/depot/332761/)
* [332762 - Friendship Club (OSX)](https://steamdb.info/depot/332762/)
* [334921 - Zero-K Content](https://steamdb.info/depot/334921/)
* [350641 - Binaries](https://steamdb.info/depot/350641/)
* [357941 - Telepath Tactics Windows](https://steamdb.info/depot/357941/)
* [358641 - Blue Estate Digital Comic Content](https://steamdb.info/depot/358641/)
* [401324 - Mighty No. 9 Mighty Month 2](https://steamdb.info/depot/401324/)
* [414261 - Ironclads 2: American Civil War Content](https://steamdb.info/depot/414261/)
* [434711 - Valley of Death Content](https://steamdb.info/depot/434711/)
* [443071 - Insane Insects: The Inception Content](https://steamdb.info/depot/443071/)
* [462241 - Post Human W.A.R Content](https://steamdb.info/depot/462241/)
* [462242 - Post Human W.A.R Mac](https://steamdb.info/depot/462242/)
* [470891 - WarFire Content](https://steamdb.info/depot/470891/)
* [470892 - WarFire English](https://steamdb.info/depot/470892/)
* [470894 - WarFire Simplified Chinese](https://steamdb.info/depot/470894/)
* [470895 - WarFire Traditional Chinese](https://steamdb.info/depot/470895/)
* [474771 - WarFire Demo Content](https://steamdb.info/depot/474771/)
* [493441 - Thanatos Content](https://steamdb.info/depot/493441/)
* [545543 - TIMODD OSX Specific](https://steamdb.info/depot/545543/)
* [545544 - TIMODD Win32 Specific](https://steamdb.info/depot/545544/)
* [545545 - TIMODD Win64 Specific](https://steamdb.info/depot/545545/)
* [545546 - TIMODD Linux Specific](https://steamdb.info/depot/545546/)
* [549401 - Ironclads 2: War of the Pacific Content](https://steamdb.info/depot/549401/)
* [560841 - Tootinis Content](https://steamdb.info/depot/560841/)
* [589671 - Double Kick Heroes Content Windows](https://steamdb.info/depot/589671/)
* [589672 - Double Kick Heroes Content MacOS](https://steamdb.info/depot/589672/)
* [596621 - Gas Guzzlers: Combat Carnage Content](https://steamdb.info/depot/596621/)
* [602861 - Ironclads 2: Boshin War Content](https://steamdb.info/depot/602861/)
* [606681 - Silver (Windows)](https://steamdb.info/depot/606681/)
* [606682 - Silver (Linux)](https://steamdb.info/depot/606682/)
* [606683 - Silver (Mac)](https://steamdb.info/depot/606683/)
* [616850 - RUNNING WITH RIFLES: The Pacific (616850) Depot](https://steamdb.info/depot/616850/)
* [616851 - RUNNING WITH RIFLES: Pacific Mac Depot](https://steamdb.info/depot/616851/)
* [629911 - Clicker Heroes 2 Windows](https://steamdb.info/depot/629911/)
* [629912 - Clicker Heroes 2 Mac OS](https://steamdb.info/depot/629912/)
* [647841 - Ironclads 2: Caroline Islands War 1885 Content](https://steamdb.info/depot/647841/)
* [671612 - Warfork Testing](https://steamdb.info/depot/671612/)
* [686473 - Zombie Panic! Source Dedicated Server Shared Content Dedicated](https://steamdb.info/depot/686473/)
* [693451 - Clad in Iron: Gulf of Mexico 1864 Content](https://steamdb.info/depot/693451/)
* [709522 - WHERE'S PHANTOM THIEF mac depo](https://steamdb.info/depot/709522/)
* [709523 - WHERE'S PHANTOM THIEF Windows depo](https://steamdb.info/depot/709523/)
* [712971 - Clad in Iron: Philippines 1898 Content](https://steamdb.info/depot/712971/)
* [751611 - SpellForce 3 Beta Content](https://steamdb.info/depot/751611/)
* [793621 - Hellbreaker Content](https://steamdb.info/depot/793621/)
* [795000 - SpellForce 3 Mod Kit (795000) Depot](https://steamdb.info/depot/795000/)
* [809711 - Fun Simulator Content](https://steamdb.info/depot/809711/)
* [813651 - RF Win BASE](https://steamdb.info/depot/813651/)
* [849041 - Double Kick Heroes Demo Content](https://steamdb.info/depot/849041/)
* [927841 - Clinically Dead Content](https://steamdb.info/depot/927841/)
* [940311 - Corpse Party 2: Dead Patient Content](https://steamdb.info/depot/940311/)
* [963681 - Clad in Iron: Sakhalin 1904 Content](https://steamdb.info/depot/963681/)
* [989581 - Windows English](https://steamdb.info/depot/989581/)
* [989583 - Windows Japanese](https://steamdb.info/depot/989583/)
* [989585 - Windows Simplified Chinese](https://steamdb.info/depot/989585/)
* [989586 - Windows Traditional Chinese](https://steamdb.info/depot/989586/)
* [1001861 - Casual Desktop Game Content](https://steamdb.info/depot/1001861/)
* [1054631 - Tank battle 个 Depot](https://steamdb.info/depot/1054631/)
* [1055431 - Swords and Sandals Classic Collection ContentMAC](https://steamdb.info/depot/1055431/)
* [1055432 - Swords and Sandals Classic Collection DepotWIN](https://steamdb.info/depot/1055432/)
* [1098271 - ColdSide Content](https://steamdb.info/depot/1098271/)
* [1098272 - Хранилище ColdSide(chinese)](https://steamdb.info/depot/1098272/)
* [1136501 - SDK Content](https://steamdb.info/depot/1136501/)
* [1144491 - Clad In Iron: Carolines 1885 Content](https://steamdb.info/depot/1144491/)
* [1161221 - Time Squared Content](https://steamdb.info/depot/1161221/)
* [1179391 - Victorian Admirals Caroline Crisis Content](https://steamdb.info/depot/1179391/)
* [1203311 - Tiles Shooter Puzzle Cube Common](https://steamdb.info/depot/1203311/)
* [1210750 - RUNNING WITH RIFLES: EDELWEISS (1210750) Depot](https://steamdb.info/depot/1210750/)
* [1210751 - RUNNING WITH RIFLES: EDELWEISS Mac Depot](https://steamdb.info/depot/1210751/)
* [1258641 - VA Marianas Incident 1887 Content](https://steamdb.info/depot/1258641/)
* [1259291 - Victorian Admirals Panama Crisis 1885 Content](https://steamdb.info/depot/1259291/)
* [1259341 - Victorian Admirals Samoan Crisis 1889 Content](https://steamdb.info/depot/1259341/)
* [1287293 - WolfTeam: Classic Mena](https://steamdb.info/depot/1287293/)
* [1293201 - Project Haven Demo Depot](https://steamdb.info/depot/1293201/)
* [1386581 - 寒刀行 Content](https://steamdb.info/depot/1386581/)
* [1416264 - SpellForce 3: Modding Tools Shared Depot](https://steamdb.info/depot/1416264/)
* [1426211 - It Takes Two - Content](https://steamdb.info/depot/1426211/)
* [1492341 - Clad in Iron: Manila Bay 1898 Content](https://steamdb.info/depot/1492341/)
* [1505701 - Berzerk Anthology Content](https://steamdb.info/depot/1505701/)
* [1593501 - God of War Content](https://steamdb.info/depot/1593501/)
* [1615101 - The Last Stand Trilogy OSX](https://steamdb.info/depot/1615101/)
* [1615102 - The Last Stand Trilogy WIN](https://steamdb.info/depot/1615102/)
* [1643701 - Visitors: First Contact Content](https://steamdb.info/depot/1643701/)
* [1653361 - "Black Space"-Depot](https://steamdb.info/depot/1653361/)
* [1659211 - Uninvited Guest Content](https://steamdb.info/depot/1659211/)
* [1674811 - into the dangerous world i leapt Content](https://steamdb.info/depot/1674811/)
* [1720221 - Aqua Boy Content](https://steamdb.info/depot/1720221/)
* [1720222 - Aqua Boy WIN](https://steamdb.info/depot/1720222/)
* [1725731 - Casual Game Engine Content](https://steamdb.info/depot/1725731/)
* [1728951 - ](https://steamdb.info/depot/1728951/)
* [1765721 - xVASynth Content](https://steamdb.info/depot/1765721/)
* [1807241 - Casual Pixel Warrior Content](https://steamdb.info/depot/1807241/)
* [1807421 - EPEJSODION Dodgeball Training Content](https://steamdb.info/depot/1807421/)
* [1858861 - Solitarius Content](https://steamdb.info/depot/1858861/)
* [1911901 - ](https://steamdb.info/depot/1911901/)
* [1922751 - xVATrainer Content](https://steamdb.info/depot/1922751/)
* [2097741 - ](https://steamdb.info/depot/2097741/)
* [2097751 - ](https://steamdb.info/depot/2097751/)
* [2114791 - ](https://steamdb.info/depot/2114791/)
* [2114792 - ](https://steamdb.info/depot/2114792/)
* [2129122 - ](https://steamdb.info/depot/2129122/)
* [2129123 - ](https://steamdb.info/depot/2129123/)
* [2196171 - ](https://steamdb.info/depot/2196171/)
* [2206761 - ](https://steamdb.info/depot/2206761/)
* [2242471 - ](https://steamdb.info/depot/2242471/)
* [2614611 - ](https://steamdb.info/depot/2614611/)
* [2614612 - ](https://steamdb.info/depot/2614612/)
* [2614613 - ](https://steamdb.info/depot/2614613/)
* [2655351 - ](https://steamdb.info/depot/2655351/)
* [2683633 - ](https://steamdb.info/depot/2683633/)
* [2683634 - ](https://steamdb.info/depot/2683634/)

Comment (MichaelHinrichs) [reply rules.ini:46]: Yeah, that's why i made a rule to find those SPECIFIC angelscript files.

Comment (Lifeismana) [reply rules.ini:46]: Then this won't get merged, looking for 158 files is too much

**Comment by xPaw** [comment]
Closing as this regex is too big.

**Comment by SteamTracker** [comment]
After running changed rules in this pull request *(this bot only tests rules.ini changes)*:

Changed technologies: **Engine.CasualGameEngine**

These apps now match: **(33)** [17500](https://steamdb.info/app/17500/depots/) [17505](https://steamdb.info/app/17505/depots/) [43190](https://steamdb.info/app/43190/depots/) [43210](https://steamdb.info/app/43210/depots/) [70900](https://steamdb.info/app/70900/depots/) [70910](https://steamdb.info/app/70910/depots/) [104310](https://steamdb.info/app/104310/depots/) [209340](https://steamdb.info/app/209340/depots/) [219830](https://steamdb.info/app/219830/depots/) [220980](https://steamdb.info/app/220980/depots/) [238320](https://steamdb.info/app/238320/depots/) [238430](https://steamdb.info/app/238430/depots/) [249950](https://steamdb.info/app/249950/depots/) [251770](https://steamdb.info/app/251770/depots/) [262100](https://steamdb.info/app/262100/depots/) [282590](https://steamdb.info/app/282590/depots/) [291570](https://steamdb.info/app/291570/depots/) [293030](https://steamdb.info/app/293030/depots/) [296770](https://steamdb.info/app/296770/depots/) [304220](https://steamdb.info/app/304220/depots/) [314710](https://steamdb.info/app/314710/depots/) [350640](https://steamdb.info/app/350640/depots/) [434710](https://steamdb.info/app/434710/depots/) [443070](https://steamdb.info/app/443070/depots/) [793620](https://steamdb.info/app/793620/depots/) [927840](https://steamdb.info/app/927840/depots/) [1001860](https://steamdb.info/app/1001860/depots/) [1653360](https://steamdb.info/app/1653360/depots/) [1725730](https://steamdb.info/app/1725730/depots/) [1807240](https://steamdb.info/app/1807240/depots/) [1858860](https://steamdb.info/app/1858860/depots/) [2567530](https://steamdb.info/app/2567530/depots/) [2655350](https://steamdb.info/app/2655350/depots/)

Please confirm that all these changed apps are correct.

**Pull request closed without merging**